src/EnergyPlus/Fans.cc

```diff
@@ -1700,7 +1701,9 @@ void FanComponent::simulateConstant(EnergyPlusData &state)
     }
 } // FanComponent::simulateConstant
 
-void FanComponent::simulateVAV(EnergyPlusData &state, ObjexxFCL::Optional<Real64 const> _pressureRise)
+void FanComponent::simulateVAV(EnergyPlusData &state,
+                               ObjexxFCL::Optional<Real64 const> _pressureRise,
+                               ObjexxFCL::Optional<Real64 const> _onOffFanPartLoadFraction)
```

Comment:
The original VAV fan model never cycled during a time step because it was used with systems that were either on the entire time step or off the entire time step. In the case of VRF, where a VAV fan is allowed for the fluid temperature control model, this fan type could cycle. For the normal VAV model (AirloopHVAC with VAV boxes) a DX coil could be used in the air loop but the fan will not cycle. In that case the DX coil could set OnOffFanPartLoadFraction to something other than 1 and the VAV fan will then multiply by this variable even though the fan is not cycling. It's not that correcting this for VRF systems is wrong, it's that using this code with normal VAV systems could lead to inaccurate fan energy. The addition of an optional argument, only for VRF, is OK since it's only used with the VRF fluid temperature control model. But this also opens up the future possibility that other developers may try to pass this argument. Also, you are passing RTF not PLF so the variable name is misleading. I'm actually not sure if using this optional argument is the better approach (compared to a warning and eventually allowing the OnOff fan with this VRF model). Maybe if you change the variable name to FanRunTimeFraction it would be a little more clear and avoid future issues. I think I would also add a comment in the header that this optional argument is only used with VRF_FluidTCtrl.

Reply:
The more I think about this the more I think the VAV fan model should not be changed. The VRF_FluidTCtrl model does allow the Fan:SystemModel which does account for cycling. So maybe a warning when using the VAV fan and the system cycles is the better approach ([like this](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/HVACVariableRefrigerantFlow.cc#L3447-L3457)). The user should use the fan system model when they expect the VRF will cycle the fan.

Reply:
I see that. I am just wondering if we should change a 20 year old fan model for this corner case or guide the user to use the correct fan model.

Reply:
Does this mean user will have to cope with the fact that cycling with variable volume fan is not possible if they used Fan:VariableVolume? If they need cycling fan cycling coil, then they should use Fan:SystemModel? And the fix in this PR will be to add some warning for the case where Fan:VariableVolume is used in cycling fan cycling coil mode?

Reply:
Maybe we can ask how people feel about these two options in the coming technicality meeting? @rraustad

Reply:
That is what I am thinking, use system model for cycling cases and VAV fan or system model for constant fan cases. In fact the system model was meant to replace most other fan types so users should get comfortable with using the system model fan exclusively.

Reply:
I see